### PR TITLE
dependency updates

### PR DESCRIPTION
**Pull request by mythi**
My 'carry-water-chop-wood' contribution to what I think is important: keep our dependencies update-to-date. This PR updates some dependencies that are know to require manual updates (i.e., `nix`) AND changes dependabot to focus on direct dependencies for now so that the "backlog" gets emptied.

### CODE REVIEW

File: attestation-agent/coco_keyprovider/Cargo.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@ base64.workspace = true
 clap = { workspace = true, features = ["derive"] }
 ctr.workspace = true
 daemonize = "0.5.0"
-env_logger = "0.10.0"
+env_logger = { workspace = true }
```
Comment (Xynnn007) [inline attestation-agent/coco_keyprovider/Cargo.toml:15]:
```suggestion
env_logger.workspace = true
```

**Review by Xynnn007 — COMMENTED** [review]
Thanks for the contribution.

File: attestation-agent/coco_keyprovider/Cargo.toml (hunk 1 above)
Comment (mythi) [reply attestation-agent/coco_keyprovider/Cargo.toml]: fixed

**Comment by mythi** [comment]
/cc @fitzthum 

it looks to me the codeowners setting does not give sufficient coverage.

**Pull request merged**